src/group_theory/order_of_element.lean

```diff
@@ -368,6 +372,23 @@ calc (univ.filter (λ a : α, a ^ n = 1)).card ≤ (gpowers (g ^ (fintype.card 
     exact le_of_dvd hn0 (gcd_dvd_left _ _)
   end
 
+section
+
+variables [group α] [fintype α] [decidable_eq α]
+
+lemma is_cyclic.image_range_order_of (ha : ∀ x : α, x ∈ gpowers a) :
```

Comment:
This is a very naïve question: why doesn't this use the `is_cyclic` typeclass?

Reply:
Because we need to talk about a generator, and the `is_cyclic` typeclass only gives you the existence of such a generator.

Reply:
I see. But it still goes under the `is_cyclic` namespace?

Reply:
Yeah, that seems to be the pattern in this file